doc/ValueExtraction_DE.md

```diff
@@ -22,40 +22,56 @@ Damit der *Smart Appliance Enabler* weiss, dass die Antwort als JSON interpretie
 
 Die Wert-Extraktion mit einem [regulären Ausdruck](http://www.regexe.de/hilfe.jsp) funktioniert immer. Allerdings erschliesst sich deren Formulierung nicht jedem sofort. 
 
-Zum Testen, ob der gewählte reguläre Ausdruck den gewünschten Wert aus der Antwort extrahiert, eignet sich dieser [Java Regex-Tester](https://www.freeformatter.com/java-regex-tester.html).
+Zum Testen, ob der gewählte reguläre Ausdruck den gewünschten Wert aus der Antwort extrahiert, eignet sich die Webseite [RegEx101](https://regex101.com/). Der Vorteil dieses Testers ist, dass er direkt während der Eingabe evaluiert.
 
 Ausser dem regulären Ausdruck benötigt man die Antwort, aus welcher der Wert extrahiert werden soll. Wenn der *Smart Appliance Enabler* bereits mit diesem Gerät kommuniziert, kann dessen Antwort dem Log entnommen werden.
 
+### Beispiel für RegEx-Anwendung auf JSON-Daten 
+
 Die kursiv dargestellten Bezeichnungen beziehen sich auf die entsprechenden Felder der Regex-Tester-Seite.
 
-_Java Regular Expression_: `.*"Power":(\d+).*`
+_Regular Expression_: `.*"Power":(\d+).*`
 
-_Entry to test against_:
+_Test Text_:
 
 ```json
 {"StatusSNS":{"Time":"2019-09-06T20:06:19","ENERGY":{"TotalStartTime":"2019-08-18T11:07:55","Total":0.003,"Yesterday":0.000,"Today":0.003,"Power":26,"ApparentPower":25,"ReactivePower":25,"Factor":0.06,"Voltage":239,"Current":0.106}}}
 ```
 
-_Replace with (Optional)_: `$1`
-
-_Flags_: `[x] Dotall`
-
-Nach Klick auf `REPLACE FIRST` wird angezeigt:
+Im rechten Teil des Fenster wird im oberen Teil direkt erklärt, was genau die einzelnen Teile der RegEx bewirken und im mittleren Teil das Ergebnis angzeigt.
 
-```
-Results
-.matches() method: true
-.lookingAt() method: true
-String replacement result:
-26
-```
+![RegEx101-Example](../pics/RegEx101-Example.png)
 
 Der Wert 26 wurde also erfolgreich mit obigem regulären Ausdruck aus der Antwort extrahiert.
 
-Wenn ein regulärer Ausdruck konfiguriert wird, sollte im *Smart Appliance Enabler* als `Format` kein Wert konfiguriert werden!
+**Wenn ein regulärer Ausdruck konfiguriert wird, sollte im *Smart Appliance Enabler* als `Format` kein Wert konfiguriert werden!**
 
 Im *Smart Appliance Enabler* muss für dieses Beispiel konfiguriert werden:
 
 `Regex für Extraktion`: `.*"Power":(\d+).*`
 
 `Format`: leer lassen
+
+### Beispiel für RegEx-Anwendung auf Key-Value-Daten
+
+Liegen die auszuwertenden Daten als Key-Value-Paare vor, so muss beachtet werden, dass der reguläre Ausdruck auf die gesamte Zeichenkette zur Anwendung kommt! Also nicht nur auf das ermittelte Value.
+
+Auch das kann mit dem [RegEx101](https://regex101.com/) gegrüft werden.
+
+Im folgenden Beispiel liegen die Zählerstände für Kühlschrank und Geschirrspüler als Key-Value-Paare vor:
+
+![MeteringKeyValueExample](../pics/MeteringKeyValueExample.png)
+
+Damit ist die für den Kühlschrank zu prüfende Zeichenkette wie folgt:
+
+_Regular Expression_: `(\d+.?\d*)`
+
+_Test String_: `kuehlschrank=123.456`
```

Comment:
Warum verwendest Du nicht die Zahl, die in `MeteringKeyValueExample.png` gezeigt wird? Das wäre anschaulicher.

Reply:
Weil ich gerade keine Beispiel direkt in kWh habe, ich aber einen Wert mit Komma beschreiben wollte. Da lass' ich lieber den Screenshot weg...

Reply:
Aktualisiert mit passendem Screenshot